plenum/persistence/req_id_to_txn.py

```diff
@@ -13,33 +13,41 @@ class ReqIdrToTxn:
     def __init__(self, keyValueStorage: KeyValueStorage):
         self._keyValueStorage = keyValueStorage
 
-    def add(self, digest, ledger_id, seq_no):
-        self._keyValueStorage.put(digest, self._create_value(ledger_id, seq_no))
-
-    def addBatch(self, batch):
-        self._keyValueStorage.setBatch([(digest, self._create_value(ledger_id,
-                                                                    seq_no))
-                                        for digest, ledger_id, seq_no in batch])
-
-    def get(self, digest):
+    def add(self, payload_digest, ledger_id, seq_no, digest):
+        self._keyValueStorage.put(payload_digest, self._create_value(ledger_id, seq_no, digest))
+
+    def addBatch(self, batch, full_digest: bool = True):
+        payload = []
+        full = []
+        for payload_digest, ledger_id, seq_no, digest in batch:
+            payload.append((payload_digest, self._create_value(ledger_id,
+                                                               seq_no,
+                                                               digest)))
+            full.append((digest, str(payload_digest)))
+        self._keyValueStorage.setBatch(payload)
+        if full_digest:
+            self._keyValueStorage.setBatch(full)
+
+    def get(self, some_digest, full_digest: bool = False):
```

Comment:
I think it would be much more clear and clean to have two methods: `get_by_payload_digest` and `get_by_full_digest`

Reply:
ok